packages/extension/src/inpage/alephiumWindowObject.ts

```diff
@@ -205,8 +211,96 @@ export const alephiumWindowObject: AlephiumWindowObject =
             salt: Date.now().toString(),
           }),
         )
-      ).result as SignUnsignedTxResult
-      return result
+      ).result
+
+      return txResult.result as SignUnsignedTxResult
+    }
+
+    signAndSubmitChainedTx = async (
+      params: SignChainedTxParams[],
+    ): Promise<SignChainedTxResult[]> => {
+      if (params.length === 0) {
+        throw Error("Empty transaction params")
+      }
+
+      const transactionParamz: TransactionParams[] = params.map(param => {
```

Comment:
[Nit]: it seems that we can use the `signedChainedTxParamsToTransactionParams` function.

Reply:
I initially used `signedChainedTxParamsToTransactionParams` and when interacting with the dApp it complained that this function can only be used by the extension (due to where its defined). I will see if there is a way to fix it.

Reply:
Fixed now, put it in the [transformer.ts](https://github.com/alephium/extension-wallet/blob/chained-txs/packages/extension/src/shared/transactions/transformers.ts#L21) instead of [index.ts](https://github.com/alephium/extension-wallet/blob/chained-txs/packages/extension/src/shared/transactions/index.ts) works.